src/main/java/org/mockito/internal/stubbing/BaseStubbing.java

```diff
@@ -24,7 +24,7 @@
     public OngoingStubbing<T> thenReturn(T value, T... values) {
         OngoingStubbing<T> stubbing = thenReturn(value);
         if (values == null) {
-            // TODO below does not seem right
+            // Return null in the 2nd and subsequent invocations.
```

Comment:
This comment explains what the code does but does not explain why (which is the point of code comments). Unfortunately, I don't know either :) I think it was a judgement call years ago and we will stick to it to keep compatibility. Thank you for cleaning up the TODO and adding test coverage.

Reply:
I see, thank you! Now I think the TODO was added because it's strange when a _user_ passes a null varargs argument:
```java
thenReturn("foo", (String []) null);
// instead of 
thenReturn("foo", (String) null);
```
If that null array comes from somewhere else (e.g., a method `nextInvocationsArguments`), that almost always a programming error that must be fixed (we can't say if a user really means `{ null }`, or {}).

Will Mockito stick with this decision in 2.x only, or in 3+ too? Will you consider changing that in the next releases? In the latter case I'd keep a TODO, but expand it with explanation _and_ a reference to an issue.

Reply:
Unless there is a strong reason not to, I plan to keep the code as is. Nobody complained about it. There is risk that the change will be incompatible. It's better to focus Mockito 3 on more interesting features!

Thank you for pointing this out and attempting to clear up the TODO!